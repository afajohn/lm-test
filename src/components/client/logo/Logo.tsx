import Image from 'next/image';

interface LogoProps {
  src: string; // Path to the logo image
  alt: string; // Alternative text for accessibility
  width?: number; // Optional width of the logo
  height?: number; // Optional height of the logo
  loading?: "eager" | "lazy";
}

const Logo: React.FC<LogoProps> = ({ src, alt, width = 100, height = 100, loading }) => {
  return (
      <Image 
        className='w-auto h-auto'
        src={src} 
        alt={alt} 
        width={width} 
        height={height} 
        loading={loading}
        // priority={true} // Ensures the logo loads quickly
      />
  );
};

export default Logo;