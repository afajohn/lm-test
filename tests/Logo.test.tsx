import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Logo from "@/components/client/logo/Logo";

describe("Logo component", () => {
	it("renders correctly", ({ expect }) => {
		const { container } = render(<Logo src="logo.png" alt="Logo" />);
		expect(container).toMatchSnapshot();
	});
	it("renders with default width and height", () => {
		const { getByRole } = render(<Logo src="logo.png" alt="Logo" />);
		const image = getByRole("img");
		expect(image).toHaveAttribute("width", "100");
		expect(image).toHaveAttribute("height", "100");
	});
	it("renders with custom width and height", () => {
		const { getByRole } = render(
			<Logo src="logo.png" alt="Logo" width={200} height={300} />,
		);
		const image = getByRole("img");
		expect(image).toHaveAttribute("width", "200");
		expect(image).toHaveAttribute("height", "300");
	});
	it("renders with alt text", () => {
		const { getByRole } = render(<Logo src="logo.png" alt="Logo" />);
		const image = getByRole("img");
		expect(image).toHaveAttribute("alt", "Logo");
	});
	it("renders with src attribute", () => {
		const { getByRole } = render(<Logo src="logo.png" alt="Logo" />);
		const image = getByRole("img");
		expect(image).toHaveAttribute("src", "logo.png");
	});
});
