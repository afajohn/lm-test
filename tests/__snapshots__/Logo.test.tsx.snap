// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo component > renders correctly 1`] = `
<div>
  <img
    alt="Logo"
    class="w-auto h-auto"
    height="100"
    loading="lazy"
    src="logo.png"
    width="100"
  />
</div>
`;
